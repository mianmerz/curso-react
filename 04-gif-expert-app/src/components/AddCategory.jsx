import { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        const text = target.value;
        setInputValue(text);
    };

    const onSubmit = (event) => {
        event.preventDefault();

        const newInputValue = inputValue.trim();

        if (newInputValue.length <= 1) return;

        onNewCategory(newInputValue);
        // setCategories(categories => [inputValue, ...categories]);
        setInputValue('');
    };

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input type="text" placeholder='Buscar gif' value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}