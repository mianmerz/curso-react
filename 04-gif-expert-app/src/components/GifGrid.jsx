import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
    console.log("category", category);

    getGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {
            }
        </>
    )
}
