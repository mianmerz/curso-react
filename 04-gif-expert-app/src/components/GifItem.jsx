
import React from 'react';
import PropTypes from 'prop-types';

export const GifItem = ({ title, url, id }) => {
  // console.log(id, title, url);

  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
