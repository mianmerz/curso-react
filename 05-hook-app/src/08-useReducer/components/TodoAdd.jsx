import { useForm } from "../../hooks/useForm";

export const TodoAdd = ({ onNewTodo }) => {

    const { description, onInputChange, onReset } = useForm({
        description: ''
    })

    const onSubmit = (e) => {
        e.preventDefault();
    
        if (description.trim().length <= 1) return;

        const newTodo = {
            id: new Date().getTime(),
            description,
            done: false
        }

        onNewTodo && onNewTodo(newTodo);
        onReset();
    }

    return (
        <form >
            <input type="text" className="form-control" placeholder="Que hay que hacer?"
                name="description" value={description} onChange={onInputChange} />

            <button type="submit" className="btn btn-primary mt-3" onClick={onSubmit}>Agregar</button>
        </form>
    )
}
