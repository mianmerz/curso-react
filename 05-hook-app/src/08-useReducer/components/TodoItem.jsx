
export const TodoItem = ({ todo, onDeleteTodo, onToggleTodo }) => {
    return (
        <li className="list-group-item d-flex justify-content-between align-items-center">
            <span
                className={"user-select-none " + (todo.done ? 'text-decoration-line-through' : '')}
                onDoubleClick={() => onToggleTodo(todo.id)}
                aria-label="Cambiar estado de la tarea"
            > {todo.description}  {todo.done && '(completa)'} </span>

            <button className="btn btn-danger" onClick={() => onDeleteTodo(todo.id)}>Borrar</button>
        </li>
    )
}
