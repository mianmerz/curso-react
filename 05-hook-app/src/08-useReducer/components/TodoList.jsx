import { TodoItem } from "./index"

export const TodoList = ({ todos = [], handleDeleteTodo }) => {
    return (
        <ul className="list-group">
            {
                todos.map(todo => <TodoItem key={todo.id} todo={todo} handleDeleteTodo={handleDeleteTodo} />)
            }

        </ul>
    )
}
