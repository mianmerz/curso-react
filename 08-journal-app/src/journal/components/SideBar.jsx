import { TurnedInNot } from "@mui/icons-material"
import { Box, Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from "@mui/material"

export const SideBar = ({ drawerWidth = 240 }) => {
    return (
        <Box
            component="nav"
            sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
        >

            <Drawer variant="permanent" open
                sx={{
                    display: { xs: 'block' },
                    '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
                }}>
                <Toolbar>
                    <Typography variant="h6" noWrap component="div">
                        Miguel Merelli
                    </Typography>
                </Toolbar>

                <Divider />

                <List>
                    {
                        ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio', 'Julio'].map(text => (

                            <ListItem key={text} dissablePadding>
                                <ListItemButton>
                                    <ListItemIcon>
                                        <TurnedInNot />
                                    </ListItemIcon>

                                    <Grid container>
                                        <ListItemText primary={text} />
                                        <ListItemText secondary={"Exercitation cillina sdlkasdlkjasd"} />
                                    </Grid>
                                </ListItemButton>
                            </ListItem>
                        ))
                    }
                    </List>
            </Drawer>
        </Box >
    )
}
