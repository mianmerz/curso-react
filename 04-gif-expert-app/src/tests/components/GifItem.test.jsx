import { render, screen } from "@testing-library/react"
import { GifItem } from "../../components/GifItem"

describe("Pruebas de GifItem", () => {
    const title = 'Saitana';
    const url = 'https://blabla.jpg';

    test('snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug()

        const { src, alt } = screen.getByRole('img');
        expect(src).toContain(url);
        expect(alt).toContain(alt);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
})