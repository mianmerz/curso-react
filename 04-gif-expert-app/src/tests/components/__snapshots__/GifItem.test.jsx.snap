// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitana"
      src="https://blabla.jpg"
    />
    <p>
      Saitana
    </p>
  </div>
</div>
`;
